Incompatibility with pytest-flakes?

After updating pytest-socket to the 0.5.0 version, I cannot use pytest-socket (0.5.0) and pytest-flakes (4.0.5) together.
I see the following error:

```
item = <FlakesItem flake-8>

    def pytest_runtest_setup(item) -> None:
        """During each test item's setup phase,
        choose the behavior based on the configurations supplied.
    
        This is the bulk of the logic for the plugin.
        As the logic can be extensive, this method is allowed complexity.
        It may be refactored in the future to be more readable.
        """
    
        # If test has the `enable_socket` marker, we accept this as most explicit.
>       if "socket_enabled" in item.fixturenames or item.get_closest_marker(
            "enable_socket"
        ):
E       AttributeError: 'FlakesItem' object has no attribute 'fixturenames'
```

The 0.4.1 version working well.

--- a/pytest_socket.py
+++ b/pytest_socket.py
@@ -112,7 +112,12 @@ def pytest_runtest_setup(item) -> None:
     This is the bulk of the logic for the plugin.
     As the logic can be extensive, this method is allowed complexity.
     It may be refactored in the future to be more readable.
+
+    If the given item is not a function test (i.e a DoctestItem)
+    or otherwise has no support for fixtures, skip it.
     """
+    if not hasattr(item, "fixturenames"):
+        return
 
     # If test has the `enable_socket` marker, we accept this as most explicit.
     if "socket_enabled" in item.fixturenames or item.get_closest_marker(

--- a/tests/test_doctest.py
+++ b/tests/test_doctest.py
@@ -1,4 +1,3 @@
-
 def test_function_with_doctest(testdir):
     testdir.makepyfile(
         '''

--- a/tests/test_doctest.py
+++ b/tests/test_doctest.py
@@ -1,0 +1,15 @@
+
+def test_function_with_doctest(testdir):
+    testdir.makepyfile(
+        '''
+        def my_sum(a, b):
+            """Sum two values.
+
+            >>> my_sum(1, 1)
+            2
+            """
+            return a + b
+        '''
+    )
+    result = testdir.runpytest("--doctest-modules")
+    result.assert_outcomes(passed=1)

